client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx

```diff
@@ -0,0 +1,221 @@
+import { Button, Gridicon } from '@automattic/components';
+import { useLocalizeUrl } from '@automattic/i18n-utils';
+import { ToggleControl, TextControl } from '@wordpress/components';
+import { localize } from 'i18n-calypso';
+import { useState, FormEvent } from 'react';
+import { connect } from 'react-redux';
+import ActionPanel from 'calypso/components/action-panel';
+import ActionPanelBody from 'calypso/components/action-panel/body';
+import ActionPanelFooter from 'calypso/components/action-panel/footer';
+import ActionPanelTitle from 'calypso/components/action-panel/title';
+import HeaderCake from 'calypso/components/header-cake';
+import Notice from 'calypso/components/notice';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import { TRANSFER_SITE } from 'calypso/lib/url/support';
+import { getCurrentUserEmail } from 'calypso/state/current-user/selectors';
+import {
+	getSelectedSiteId,
+	getSelectedSite,
+	getSelectedSiteSlug,
+} from 'calypso/state/ui/selectors';
+import { useStartSiteOwnerTransfer } from './use-start-site-owner-transfer';
+
+type Props = {
+	currentUserEmail: string | null;
+	selectedSiteId: number | null;
+	selectedSiteSlug: string | null;
+	selectedSiteTitle: string | undefined;
+	translate: ( text: string, args?: Record< string, unknown > ) => string;
+};
+
+const StartSiteOwnerTransfer = ( {
+	currentUserEmail,
+	selectedSiteId,
+	selectedSiteSlug,
+	selectedSiteTitle,
+	translate,
+}: Props ) => {
+	const localizeUrl = useLocalizeUrl();
+	const [ confirmFirstToggle, setConfirmFirstToggle ] = useState( false );
+	const [ confirmSecondToggle, setConfirmSecondToggle ] = useState( false );
+	const [ newOwnerUsername, setNewOwnerUsername ] = useState( '' );
+	const [ startSiteTransferError, setStartSiteTransferError ] = useState( '' );
+	const [ startSiteTransferSuccess, setStartSiteTransferSuccess ] = useState( false );
+
+	const { startSiteOwnerTransfer, isLoading: isStartingSiteTransfer } = useStartSiteOwnerTransfer(
+		selectedSiteId,
+		{
+			onMutate: () => {
+				setStartSiteTransferError( '' );
+				setStartSiteTransferSuccess( false );
+			},
+			onError: ( error ) => {
+				setStartSiteTransferError( error.message );
+			},
+			onSuccess: () => {
+				setStartSiteTransferSuccess( true );
+			},
+		}
+	);
+
+	const handleFormSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+		startSiteOwnerTransfer( { newSiteOwner: newOwnerUsername } );
+	};
+
+	const startSiteTransferForm = (
```

Comment:
What do you think about moving the form to the separate child component? Or if we are not going to reuse it anywhere, or use in storybook, maybe we can keep the form inline with the rest of component code?

Reply:
@wojtekn The form is changing in a follow-up issue https://github.com/Automattic/dotcom-forge/issues/2496#issuecomment-1558897012